### PR TITLE
Refactor agent

### PR DESCRIPTION
**Pull request by Atralupus**
- [x] split miner
- [x] split swarm runner
- [x] add default mine handler(tx retry)

### CODE REVIEW

File: planet-clicker/Assets/LibplanetUnity/DefaultMineHandler.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            {
+                Agent.Instance.MakeTransaction((IImmutableList<ActionBase>) retryAction);
+            }
+        }
+        public void Success(Block<PolymorphicAction<ActionBase>> block)
+        {
+            Debug.Log($"created block index: {block.Index}, difficulty: {block.Difficulty}");
+        }
+    }
+}
```
Comment (greymistcube) [inline planet-clicker/Assets/LibplanetUnity/DefaultMineHandler.cs]: Needs an EOL fix 😶

File: planet-clicker/Assets/LibplanetUnity/MonoSingleton.cs [hunk 2]
```diff
@@ -1,65 +1,81 @@
...
+    /// <typeparam name="T">The type.</typeparam>
+    public abstract class MonoSingleton<T> : MonoBehaviour
+        where T : MonoSingleton<T>
     {
-        private static T _instance;
-        
         private static readonly object Lock = new object();
-        private static bool _applicationIsQuitting;
+        private static T _instance = null;
+        private static bool _appLicationIsQuitting;
```
Comment (greymistcube) [inline planet-clicker/Assets/LibplanetUnity/MonoSingleton.cs]:
```suggestion
        private static bool _applicationIsQuitting;
```

File: planet-clicker/Assets/LibplanetUnity/SwarmRunner.cs [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                    ByteUtil.Hex(PrivateKey.PublicKey.Format(true)),
+                    _swarm.EndPoint.Host,
+                    _swarm.EndPoint.Port
+                );
+            });
+
+            yield return new WaitUntil(() => swarmStartTask.IsCompleted);
+        }
+    }
+}
```
Comment (greymistcube) [inline planet-clicker/Assets/LibplanetUnity/SwarmRunner.cs]: EOL 😶

**Review by greymistcube — APPROVED** [review]

File: planet-clicker/Assets/LibplanetUnity/SwarmRunner.cs (hunk 3 above)
Comment (Atralupus) [reply planet-clicker/Assets/LibplanetUnity/SwarmRunner.cs]: 😭 I need lint...

**Comment by Atralupus** [comment]
🚀

**Pull request merged**